### PR TITLE
Bugfix Handle Temp Encrypted File Correctly

### PR DESCRIPTION
**Pull request by alperozturk96**
- [x] Tests written, or not not needed

1. An encrypted file is temporarily generated in the application directory instead of being stored directly on the device.
2. The temporarily encrypted file is automatically deleted under any circumstances during the uploading process.
3. The filename is correctly displayed in the notification.
4. When user clear failed uploads or clear cancelled uploads, temp_encrypted directory will be deleted as well.

**Note**

cacheDir can be used as well for this operation application directory chosen instead cacheDir just in case to cover edge case such as upload very big file can be failed due to cache directory

java.io.tmpdir doesn't have automatically remove mechanism or provide different solution to than File.createTempFile(). It's just a default temporary-file directory. For deletion on when virtual machine killed, **deleteOnExit()** method must be used. And you can use deleteOnExit() with any file created via createTempFile().

In our case, it will not save us from crash during upload scenario as well. 

Please read [documentation](https://docs.oracle.com/javase/8/docs/api/java/io/File.html) before PR Review. @ZetaTom @JonasMayerDev 

**Known Issues and Scope of PR**

Resuming or retrying cancelled uploads done via OCUpload therefore even our deleted temp encrypted files are not affect to the retry or resume scenario because we can get fully encrypted or decrypted path from OCUpload for retry or resume operation.

However during my tests e2e uploads and normal uploads for retry scenario worked but not always so for this PR it's out of scope. Please understand the general purpose of this PR, then review.

### CODE REVIEW

File: app/src/main/java/com/owncloud/android/operations/UploadFileOperation.java [hunk 1]
```diff
@@ -442,6 +442,7 @@ private RemoteOperationResult encryptedUpload(OwnCloudClient client, OCFile pare
         File temporalFile = null;
```
Comment (alperozturk96) [inline app/src/main/java/com/owncloud/android/operations/UploadFileOperation.java:442]: Removing usage of temporal file breaks to catch scenario

**Review by tobiasKaminsky — APPROVED** [review]

**Comment by skjnldsv** [comment]
Was checking something with the backport bot, nothing to see here, move along 
Awkward Season 4 GIF by The Office

**Pull request merged**